libs/store/src/lib/reducers/add.ts

```diff
@@ -0,0 +1,83 @@
+import { AppStateModel, AppTournamentId } from '@razor/models';
+import { extractId, extractIdType } from '@razor/util';
+import {
+  addPlayerReducerPayload,
+  addRaceReducerPayload,
+  addTournamentReducerPayload,
+} from '../payloads';
+
+// Basic Add Operations
+export const addTournamentReducer = (
+  state: AppStateModel,
+  payload: addTournamentReducerPayload,
+): AppStateModel => {
+  const { tournamentId, tournament } = payload;
+  const newState: AppStateModel = {
+    ...state,
+    tournamentsModel: {
+      ...state.tournamentsModel,
+      [tournamentId]: tournament,
+    },
+  };
+  return newState;
+};
+
+export const addRaceReducer = (
+  state: AppStateModel,
+  payload: addRaceReducerPayload,
+): AppStateModel => {
+  const { raceId, race } = payload;
+  const tournamentId: AppTournamentId = extractId(
+    raceId,
+    extractIdType.race,
+    extractIdType.tournament,
+  ) as AppTournamentId;
```

Comment:
Can you write the function to return the relevant type based on the parameters passed in, instead of doing the type cast here?

Reply:
Fixed in PR #14 under commit 42e5c196dd4d85bfe7c232dd718c2b5c5935231e